modules/phase_field/src/postprocessors/GrainTracker.C

```diff
@@ -308,15 +308,30 @@ GrainTracker::trackGrains()
         mooseError("Error with ESBD Mapping (see above unused indices)");
       }
     }
-    else // Just assign IDs in order
+    else
     {
+      /**
+       * Here we want to assign the grains in some partitioning invariant way. We'll sort first by
+       * _var_idx (already partitioned in that manner) then sort on the _min_entity_id
+       * to order the grains.
+       */
       unsigned int counter = 0;
       for (auto map_num = decltype(_maps_size)(0); map_num < _maps_size; ++map_num)
```

Comment:
Oh - well now I remember part of the reason I didn't go with that syntax. It's because the actual type for many of these counters is const! `_maps_size` is `const unsigned int`. Therefore the code won't compile without a really ugly const cast! I searched for awhile and the syntax I have now was one of the better ones I encountered.